mailet/standard/src/test/java/org/apache/james/transport/matchers/HasAttachmentTest.java

```diff
@@ -0,0 +1,108 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.matchers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Properties;
+
+import javax.mail.Session;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.base.test.FakeMail;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public class HasAttachmentTest {
+
+    private HasAttachment testee;
+    private MimeMessage mimeMessage;
+    private Mail mail;
+
+    @Before
+    public void setUp() throws Exception {
+        testee = new HasAttachment();
+
+        mimeMessage = new MimeMessage(Session.getDefaultInstance(new Properties()));
+        mail = new FakeMail();
+        mail.setRecipients(ImmutableList.of(new MailAddress("me@james.apache.org")));
+        mail.setMessage(mimeMessage);
+    }
+
+    @Test
+    public void textMailsShouldNotBeMatched() throws Exception {
+        mimeMessage.setText("A simple text message");
+
+        assertThat(testee.match(mail)).isNull();
+    }
+
+    @Test
+    public void emptyMultipartShouldNotBeMatched() throws Exception {
+        mimeMessage.setContent(new MimeMultipart());
+
+        assertThat(testee.match(mail)).isNull();
+    }
+
+    @Test
+    public void inlinedOnlyMultipartShouldNotBeMatched() throws Exception {
```

Comment:
Not 100% sure, but I believe it is more pertinant than using filename, which serves a different purpose.

So far, in the mail I received, inline is used a lot for : 
 - mail signature
 - html support
 - adding footer for mailing lists

Systematically though, the filename parameter is not mentionned in these cases. However, I think I should be able to add a filename in these cases (ex filename="footer.txt") without modifying mail server behaviour.

That being said, regarding RFC-2183 : 

> Two values for this  header field are described in this memo; one for the ordinary linear presentation of the body part, and another to facilitate the use of  mail to transfer files.

> Two common ways of presenting multipart electronic messages are as a main document with a list of separate attachments, and as a single  document with the various parts expanded (displayed) inline.

Clearly Content-Disposition is here to point out if the Mime part is an attachment.

Now regarding filenames : 

> In addition to allowing the sender to specify the presentational disposition of a message component, it is desirable to allow her to indicate a default archival disposition; a filename. The optional "filename" parameter provides for this.

It serves an other purpose.